horreum-web/src/Login.tsx

```diff
@@ -38,12 +38,18 @@ export default function LoginModal(props: LoginModalProps) {
                 onClick={() => {
                     setCreating(true)
                     store.dispatch({type: BASIC_AUTH, username, password});
-
-                    // TODO: instead of fetching userdata, should be fetching the user roles instead
-                    userApi.info([username || ''])
-                        .then(userdata => {
-                                alerting.dispatchInfo("LOGIN", "Log in successful", "Successful log in of user " + userdata[0].username, 3000)
-                                store.dispatch({type: UPDATE_ROLES, authenticated: true, roles: [/* TODO: the roles fetched */]});
+                    
+                    userApi.getRoles()
+                        .then(roles => {
+                                alerting.dispatchInfo("LOGIN", "Log in successful", "Successful log in of user " + username, 3000)
+                                store.dispatch({type: UPDATE_ROLES, authenticated: true, roles: roles});
+                                userApi.searchUsers(username!).then(
+                                    userData => store.dispatch({type: STORE_PROFILE, profile: userData.filter(u => u.username == username).at(0)}),
+                                    error => alerting.dispatchInfo("LOGIN", "Unable to get user profile", error, 30000))
+                                userApi.defaultTeam().then(
+                                    response => store.dispatch({ type: UPDATE_DEFAULT_TEAM, team: response }),
+                                    error => alerting.dispatchInfo("LOGIN", "Cannot retrieve default team", error, 30000)
+                                )
```

Comment:
What is the definition of a "successful" login?  we need all the user info for the UI to function properly, so if the call to `userApi.defaultTeam()` fails, I don't think we should display a "Log in successful" message.

At the moment the user could see;
"Successful log in of user " msg
followed by
"Cannot retrieve default team" msg

I think this would be confusing.

I think we need to chain the api calls in .then()s and only display "Login Successful" if all the calls do not error

Reply:
good question. a successful login would be to get `200 OK` response from the `roles` endpoint. the user profile and default team are optional in the frontend.
 
most likely, we'll have to revisit the login procedure in the near future.